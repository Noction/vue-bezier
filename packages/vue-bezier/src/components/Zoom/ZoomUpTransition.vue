<template>
  <component
    :is="componentType"
    :tag="props.tag"
    name="noc-zoom-up"
    v-bind="{ ...$attrs, ...hooks }"
  >
    <slot />
  </component>
</template>

<script setup lang="ts">
import type { ComponentEvents, ComponentProps } from '@/types'
import { buildComponentType, useHooks } from '@/composables'

defineOptions({ inheritAttrs: false })

const props = withDefaults(defineProps<ComponentProps>(), {
  delay: 0,
  duration: 300,
  group: false,
  origin: '',
  styles: () => ({
    animationFillMode: 'both',
    animationTimingFunction: 'ease-out'
  }),
  tag: 'span'
})

const emit = defineEmits<ComponentEvents>()

const componentType = buildComponentType(props)

const hooks = useHooks(props, emit)

</script>

<style>
  @keyframes zoom-in-up {
    0% {
      opacity: 0;
      transform: scaleY(0);
      transform-origin: top center;
    }

    50% { opacity: 1; }

    100% { transform: scaleY(1); }
  }

  @keyframes zoom-out-up {
    0% { opacity: 1; }

    50% { transform: scaleY(0); }

    100% { opacity: 0; }
  }

  .noc-zoom-up-enter-active { animation-name: zoom-in-up; }
  .noc-zoom-up-leave-active { animation-name: zoom-out-up; }
  .noc-zoom-up-move { transition: transform .3s ease-out; }
</style>
