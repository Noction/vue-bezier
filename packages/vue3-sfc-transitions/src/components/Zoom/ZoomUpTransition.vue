<template>
  <component
    :is="componentType"
    :tag="tag"
    v-bind="{ ...$attrs, ...hooks }"
    enter-active-class="zoom-in-up"
    move-class="zoom-move"
    leave-active-class="zoom-out-up"
  >
    <slot />
  </component>
</template>

<script lang="ts">
export const customProps = {
  styles: {
    default: () => {
      return {
        animationFillMode: 'both',
        animationTimingFunction: 'ease-out'
      }
    },
    type: Object
  }
}

export default {
  inheritAttrs: false
}
</script>

<script setup lang="ts">
import {
  buildComponentType,
  buildEmits,
  buildHooks,
  buildProps,
  buildTag
} from '../../composable'

const props = defineProps(buildProps(customProps))
const emit = defineEmits(buildEmits())

const componentType = buildComponentType(props)
const tag = buildTag(props)
const hooks = buildHooks(props, emit)

</script>

<style lang="scss">
  @import 'move';

  @keyframes zoom-in-up {
    0% {
      opacity: 0;
      transform: scaleY(0);
      transform-origin: top center;
    }

    50% { opacity: 1; }

    100% { transform: scaleY(1); }
  }

  @keyframes zoom-out-up {
    0% { opacity: 1; }

    50% { transform: scaleY(0); }

    100% { opacity: 0; }
  }

  .zoom-in-up { animation-name: zoom-in-up; }
  .zoom-out-up { animation-name: zoom-out-up; }
</style>
