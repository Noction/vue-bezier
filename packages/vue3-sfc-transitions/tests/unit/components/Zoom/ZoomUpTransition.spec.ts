import ZoomUpTransition from '../../../../src/components/Zoom/ZoomUpTransition.vue'
import { mount } from '@vue/test-utils'

describe('components / Zoom / ZoomUpTransition', () => {
  it('When mounting component with default props, it should work fine', () => {
    const wrapper = mount(ZoomUpTransition)

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('When mounting component with group enabled prop, it should work fine', () => {
    const wrapper = mount(ZoomUpTransition, { props: { group: true } })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
