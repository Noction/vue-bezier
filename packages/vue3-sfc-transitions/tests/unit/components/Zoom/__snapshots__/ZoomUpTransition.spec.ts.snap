// Vitest Snapshot v1

exports[`components / Zoom / ZoomUpTransition > When mounting component with default props, it should work fine 1`] = `"<transition-stub></transition-stub>"`;

exports[`components / Zoom / ZoomUpTransition > When mounting component with group enabled prop, it should work fine 1`] = `"<transition-group-stub></transition-group-stub>"`;
